http://sourceware.org/ml/gdb-cvs/2010-10/msg00070.html

### src/gdb/testsuite/ChangeLog	2010/10/11 08:47:47	1.2473
### src/gdb/testsuite/ChangeLog	2010/10/11 21:36:25	1.2474
## -1,5 +1,10 @@
 2010-10-11  Jan Kratochvil  <jan.kratochvil@redhat.com>
 
+	* gdb.cp/infcall-dlopen.cc (openlib): Support NULL FILENAME.
+	(main): Make openlib dummy call.
+
+2010-10-11  Jan Kratochvil  <jan.kratochvil@redhat.com>
+
 	Fix missing _start PIE relocation on ppc64 due to stop on dl_main.
 	* break-interp.exp (reach): Move the core body ...
 	(reach_1): ... here.  Use stop-on-solib-events for _dl_debug_state.
--- src/gdb/testsuite/gdb.cp/infcall-dlopen.cc	2010/08/17 21:31:13	1.1
+++ src/gdb/testsuite/gdb.cp/infcall-dlopen.cc	2010/10/11 21:36:26	1.2
@@ -23,6 +23,9 @@
 {
   void *h = dlopen (filename, RTLD_LAZY);
 
+  if (filename == NULL)
+    return 0;
+
   if (h == NULL)
     return 0;
   if (dlclose (h) != 0)
@@ -33,5 +36,8 @@
 int
 main (void)
 {
+  /* Dummy call to get the function always compiled in.  */
+  openlib (NULL);
+
   return 0;
 }
